jujugui/static/gui/src/app/app.js

```diff
@@ -992,25 +961,26 @@ YUI.add('juju-gui', function(Y) {
         console.error(`no linkContainerId: ${linkContainerId}`);
         return;
       }
-      const bakeryFactory = this.bakeryFactory;
       const charmstore = this.get('charmstore');
+      const bakery = this.bakery;
       const USSOLoginLink = (<window.juju.components.USSOLoginLink
-          displayType={'text'}
-          loginToController={controllerAPI.loginWithMacaroon.bind(
-            controllerAPI, bakeryFactory.get('juju'))}
-        />);
+        displayType={'text'}
+        loginToController={controllerAPI.loginWithMacaroon.bind(
+          controllerAPI, bakery)}
+      />);
+
       const LogoutLink = (<window.juju.components.Logout
-          logout={this.logout.bind(this)}
-          clearCookie={bakeryFactory.clearAllCookies.bind(bakeryFactory)}
-          gisfLogout={window.juju_config.gisfLogout || ''}
-          gisf={window.juju_config.gisf || false}
-          charmstoreLogoutUrl={charmstore.getLogoutUrl()}
-          getUser={this.getUser.bind(this, 'charmstore')}
-          clearUser={this.clearUser.bind(this, 'charmstore')}
-          // If the charmbrowser is open then don't show the logout link.
-          visible={!this.state.current.store}
-          locationAssign={window.location.assign.bind(window.location)}
-        />);
+        logout={this.logout.bind(this)}
+        clearCookie={bakery.storage.clear.bind(bakery.storage, () => {})}
```

Comment:
The bakery storage contract does not accept the callback anymore, and this is a leftover from my original impl.
So now this should be just `bakery.storage.clear.bind(bakery.storage)`.